### DISCUSSION TITLE
React18 recoilエラー

### DISCUSSION THREAD

**Discussion by takosan8** [if-mentor/next_fire_todo_5#57] [Q&A]
issue #54 
にて、issue内に添付した画像のようなエラーが発生しています。
ある程度調べたのですが、React18にrecoilのバージョンを合わせた以外は
原因や解決方法にたどり着けていません。

何かわかる方いらっしゃれば教えていただけるとありがたいです。

**Comment by Junkichi89** [comment]
単純にReactのバージョンを下げてしまうのが、一番手っ取り早い解決法かなと思いますが、どうでしょうか？

無理矢理にでも使っていくといのであれば、以下の記事を参考にして、useEffectを使って行くという手もありますが、
プロジェクト的に管理しづらくなりそうかなと思います。

https://zenn.dev/takewell/articles/5ee9530eedbeb82e4de7
https://qiita.com/horihorikei/items/4c467009a596d24949b2 これの②も同じかな


これならいい感じに行けるかな？
https://scrapbox.io/miyamonz/Next.js%E3%82%92SSR%E3%81%A8%E3%81%8B%E5%88%87%E3%81%A3%E3%81%A6SPA%E3%81%AE%E3%82%88%E3%81%86%E3%81%AB%E4%BD%BF%E3%81%86

パターン2でいけばエラー解消できると思います！
今回は、SSR使う予定ないので、これでいいと思います。
部分的にSSRつかう必要があるならその時に考えましょうって感じでもいっかも

https://ryotarch.com/javascript/react/next-js-with-csr/

**Comment by takosan8** [comment]
@Junkichi89 
解決方法のご提示ありがとうございます！
個人的にはReactを17系に下げてしまうのがいいかと思っています。
理由としてはv18起因のエラーが発生した際に、解消事例を見つけにくく、比較的対応に時間を要してしまうと思うからです。
他の方の意見も参考にしたいのでslackの方でも聞いてみようと思います。

**Reply by Yukitkym** [reply]
@takosan8
Reactを17系に下げるのに賛成です！
18系のままだとエラーが出た際にも、18系関連のエラーなのかコードの内容のミスによるエラーなのかの判別がすぐにできず、エラーの解決に時間がかかってしまい開発効率が下がってしまうかなと思いました。

**Reply by ryo0705s** [reply]
@takosan8 
recoilを導入する事に時間をかけたくない　という前提で話をするのであれば、17系に下げた方が良いと思います。
ただ、”18系で導入するという”　事をやり切ることにより、
・reactやrecoilやその周辺に付随する知識を増やすことができる
・18系でrecoilを導入する時に注意すること　的な記事を書くことができる
というメリットもあるかと思います。

**Comment by takosan8** [comment]
@Yukitkym @ryo0705s 
ご意見ありがとうございます！
たしかに18系でやりきる場合のメリットもありますね。
開発メンバーそれぞれの意向もあるかと思いますので、discussionで判断しきれない場合は次回MTGで決議とろうと思います。
